### PR TITLE
chap0 submission final--

### PR DESCRIPTION
**Pull request by sritej-reddy**
PR for merging into master (created a separate folder as said in the issue)

### CODE REVIEW

File: c-basics/src/isOlder.cpp [hunk 1]
```diff
@@ -44,52 +44,52 @@ bool validatedate(char *dob1, char *dob2, int *d1, int *d2)
...
-								else if ((*y2 % 4 != 0) && *d2>28)
-									return false;
-							}
-								return true;
-						}
-						else return false;
+		*y2 = (((((((*(dob2 + 6) - 48) * 10) + (*(dob2 + 7) - 48)) * 10) + (*(dob2 + 8) - 48)) * 10) + *(dob2 + 9) - 48);
+		if ((*d1 > 31 || *d2 > 31) || (*m1 > 12 || *m2 > 12))
+			return false;
+		if (*m1 == 2)
```
Comment (krushiraj) [inline c-basics/src/isOlder.cpp]: I have asked you to make this modular. Can you please commit again after making those changes. No need of creating new pull requests all commits will be automatically tracked.

**Review by krushiraj — CHANGES_REQUESTED** [review]
Almost everything is good except for that DOB comparison function. Please make it modular and commit. After that, we'll merge this.

**Comment by krushiraj** [comment]
Add comments or description to automatically close and tag the issue I have raised.

File: c-basics/src/isOlder.cpp (hunk 1 above)
Comment (sritej-reddy) [reply c-basics/src/isOlder.cpp]: I commited with requested changes. Closes #3

File: c-basics/src/isOlder.cpp [hunk 2]
```diff
@@ -41,89 +41,95 @@ bool validatedate(char *dob1, char *dob2, int *d1, int *d2)
...
+					}
+					else if (date2 > date1)
+					{
+						return 1;
+					}
+					else return 0;
+				}
 			}
 		}
-
```
Comment (krushiraj) [inline c-basics/src/isOlder.cpp]: Looks good for now. From next make sure that the code looks as modular as possible.

**Review by krushiraj — APPROVED** [review]

**Pull request merged**